File: gaia-sdk-javascript/src/graphql/response/type/Retrieval.ts

import {Experience} from "./Experience";
import {Knowledge} from "./Knowledge";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

export interface Retrieval {
    /**
    * Container element which collects all information static data
    */
    knowledge?:Knowledge, 
    /**
    * Container element which collects all information about runtime data
    */
    experience?:Experience
}

File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/Retrieval.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

class Retrieval: Type() {

    /**
     * Container element which collects all information static data
     */
    fun knowledge(config: Knowledge.() -> Unit) = 
        add { "knowledge{ " + Knowledge().apply(config).render(it) + "}"}


    /**
     * Container element which collects all information about runtime data
     */
    fun experience(config: Experience.() -> Unit) = 
        add { "experience{ " + Experience().apply(config).render(it) + "}"}

}

